### PR TITLE
config: move --ticketaddress to --ticketbuyer.votingaddress

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

**Comment by jrick** [comment]
I'd like to see this option renamed to votingaddress instead of ticketaddress, since it is the address with voting rights and the ticket purchase has many addresses in it.

File: config.go [hunk 1]
```diff
@@ -889,6 +891,18 @@ func loadConfig() (*config, []string, error) {
 			oldTBConfigFile, configFilePath)
 	}
 
+	// Warn if user still is still using --ticketaddress
+	if cfg.TicketAddress != nil {
```
Comment (jrick) [inline config.go]: wrong check, this is always true (cfg.TicketAddress is a non-nil AddressFlag)

File: config.go [hunk 2]
```diff
@@ -151,27 +150,29 @@ type config struct {
...
+	MaxPriceAbsolute          *cfgutil.AmountFlag  `long:"maxpriceabsolute" description:"Maximum absolute price to purchase a ticket"`
+	MaxPriceRelative          float64              `long:"maxpricerelative" description:"Scaling factor for setting the maximum price, multiplies by the average price"`
+	BalanceToMaintainAbsolute *cfgutil.AmountFlag  `long:"balancetomaintainabsolute" description:"Amount of funds to keep in wallet when stake mining"`
+	BalanceToMaintainRelative float64              `long:"balancetomaintainrelative" description:"Proportion of funds to leave in wallet when stake mining"`
+	NoSpreadTicketPurchases   bool                 `long:"nospreadticketpurchases" description:"Do not spread ticket purchases evenly throughout the window"`
+	DontWaitForTickets        bool                 `long:"dontwaitfortickets" description:"Don't wait until your last round of tickets have entered the blockchain to attempt to purchase more"`
+	VotingAddress             *cfgutil.AddressFlag `long:"votingaddress" description:"Send all ticket outputs to this address (P2PKH or P2SH only)"`
```
Comment (jrick) [inline config.go]: this description could be improved, e.g. "Purchase tickets with voting rights assigned to this address"

File: config.go [hunk 3]
```diff
@@ -889,6 +891,18 @@ func loadConfig() (*config, []string, error) {
 			oldTBConfigFile, configFilePath)
 	}
 
+	// Warn if user still is still using --ticketaddress
+	if cfg.TicketAddress.Address != nil {
+		log.Warnf("--ticketaddress has been DEPRECATED.  Use " +
+			"--ticketbuyer.votingaddress instead")
+		if cfg.TBOpts.VotingAddress.Address == nil {
+			cfg.TBOpts.VotingAddress = cfg.TicketAddress
```
Comment (jrick) [inline config.go]: I really don't like reassigning the flag.  Can you create a local var for the address instead and use that below?

File: config.go [hunk 4]
```diff
@@ -889,6 +891,18 @@ func loadConfig() (*config, []string, error) {
...
+		} else {
+			log.Warnf("Using %v as voting address",
```
Comment (jrick) [inline config.go]: don't think we need this warning

**Comment by dajohi** [comment]
addressed review items...

**Pull request merged**